config/initializers/redis.rb

```diff
@@ -0,0 +1,4 @@
+require "redis"
+
+uri = URI.parse(ApplicationConfig["REDISCLOUD_URL"])
```

Comment:
shouldn't this backup to `REDIS_URL` if `REDISCLOUD_URL` is empty? Thinking of generalization purposes

Reply:
REDISCLOUD_URL has been given a default value in the Envfile so I didnt feel the need to provide a second fallback.

Reply:
Maybe it's the naming that is bugging at me, though I understand it's due to the addon deciding the name for us...

The name of the variable sounds like we're requiring Redis Cloud as a product, but instead we're only asking for _a_ Redis URL.

Don't know, it's a little bit weird

Reply:
I agree here @rhymes in principle, but there are other ways we handle similar things that aren't necessarily cleaner...

For example, `carrierwave.rb`.

```ruby
CarrierWave.configure do |config|
  if Rails.env.development? || Rails.env.test?
    config.storage = :file
    config.enable_processing = false
  else
    config.fog_provider = "fog/aws"
    config.fog_credentials = {
      provider: "AWS",
      aws_access_key_id: ApplicationConfig["AWS_ID"],
      aws_secret_access_key: ApplicationConfig["AWS_SECRET"],
      region: "us-east-1"
    }
    config.fog_directory = ApplicationConfig["AWS_BUCKET_NAME"]
    config.storage = :fog
  end
end
```

We could maybe add `Rails.env.development?` somewhere but that might just junk things up. Maybe the solution here is a comment to clarify what's going on here. Eventually we want to work out all these funny true or implied vendor dependencies in our code, but wherever the issue is more cosmetic than structural, I'd lean towards being cool with this sort of thing and adding code comments to describe the justification.

Reply:
Ok, I see what you are saying, let me try some things to make it more clear there is no vendor dependency here.

Reply:
How would you both feel about something like this?
```ruby
# DEV uses the RedisCloud Heroku Add-On which comes with the predefiend env variable REDISCLOUD_URL
redis_url = ApplicationConfig["REDISCLOUD_URL"]
redis_url ||= ApplicationConfig["REDIS_URL"]

uri = URI.parse(redis_url)
RedisClient = Redis.new(host: uri.host, port: uri.port, password: uri.password)
```

Reply:
I like it. Though I think `ApplicationConfig["REDISCLOUD_URL"]` should be `Env["REDISCLOUD_URL"]` because ApplicationConfig refers only to the `application.yml` + `Envfile` combo